packages/synapse-sdk/src/telemetry/shutdown-utils.ts

```diff
@@ -0,0 +1,54 @@
+// telemetry/shutdown.ts
+import { removeGlobalFetchWrapper } from './fetch-wrapper.ts'
+import type { Sentry as SentryType } from './get-sentry.ts'
+
+export function setupShutdownHooks(sentry: SentryType, opts: { timeoutMs?: number } = {}) {
+  const g = globalThis as any
+  const timeout = opts.timeoutMs ?? 2000
+  let shuttingDown = false
+
+  // NOTE: sentry handles uncaughtException and unhandledRejection. see https://docs.sentry.io/platforms/javascript/troubleshooting/#third-party-promise-libraries
+
+  if (typeof g.window !== 'undefined' && typeof g.document !== 'undefined') {
+    // -------- Browser runtime --------
+    const flush = () => {
```

Comment:
Why do we only flush in browsers vs. close?  Lets document that.  Is there any harm in just closing?